src/main/java/seedu/duke/Duke.java

```diff
@@ -0,0 +1,188 @@
+package seedu.duke;
+
+import seedu.duke.commands.Command;
+import seedu.duke.commands.ListCommand;
+import seedu.duke.commands.AddCommand;
+import seedu.duke.commands.DoneCommand;
+import seedu.duke.commands.ByeCommand;
+import seedu.duke.commands.DeadlineCommand;
+import seedu.duke.commands.EventCommand;
+import seedu.duke.commands.TodoCommand;
+
+
+import seedu.duke.exceptions.InsufficientArgumentException;
+import seedu.duke.exceptions.InvalidArgumentException;
+import seedu.duke.exceptions.InvalidCommandException;
+import seedu.duke.trackables.Deadline;
+import seedu.duke.trackables.Event;
+import seedu.duke.trackables.Task;
+import seedu.duke.trackables.Todo;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Duke {
```

Comment:
Regarding Duke class, maybe you want to separate the functionalities in it into more classes? Like Storage, Parser etc. I think there are too many functionalities in the Duke class and it doesn't really correspond to the Single Responsibility principle in SOLID. It may make the process of changing the program's code to fit new requirement harder as it may be quite hard to locate the right piece of code to change among the extensive lines of code.

Also, I think the purpose of Duke class should act like a body which links all its parts(Storage, Parser, TaskList) together and not doing too much of logic. This concept may help you to refactor your code.